File: src/HeuristicOptimizationTechniques/Helper/Location.java
package HeuristicOptimizationTechniques.Helper;

public class Location {
    private int x;
    private int y;

    public Location(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    private int distance(Location other) {
        double dx = this.x  - other.x;
        double dy = this.y - other.y;
        return (int) Math.ceil(Math.sqrt(dx * dx + dy * dy));
    }

    @Override
    public String toString() {
        return "(" + x + ", " + y + ")";
    }
}

File: src/HeuristicOptimizationTechniques/Helper/Location.kt
package HeuristicOptimizationTechniques.Helper

import kotlin.math.ceil
import kotlin.math.sqrt

data class Location(var x: Int, var y: Int, var index: Int) {
    override fun toString(): String {
        return "(index: $index, coords: ($x, $y))"
    }

    fun distance(b: Location): Int {
        val dx = (x - b.x).toDouble()
        val dy = (y - b.y).toDouble()
        return ceil(sqrt(dx * dx + dy * dy)).toInt()
    }

    companion object {
        fun distWithoutDepot(route: List<Location>): Int =
            route.zipWithNext().sumOf { (a, b) -> a.distance(b) }
    }

}
